pkg/pods/pod.go

```diff
@@ -1057,3 +1059,32 @@ func (p *Pod) withTimeWarnings(scriptType string, serviceID string, f func()) {
 
 	f()
 }
+
+func (p *Pod) runWithTerminationGracePeriod(scriptType string, serviceID string, f func()) {
+	doneCh := make(chan struct{})
+	go func() {
+		f()
+		close(doneCh)
+	}()
+
+	manifest, err := p.CurrentManifest()
+	if err != nil {
+		p.logger.WithError(err).Errorln("This pod does not have a manifest, cannot determine its config file")
+	}
+	gracePeriod := TerminationGracePeriod
+	if manifest.GetTerminationGracePeriod() != 0 {
+		gracePeriod = manifest.GetTerminationGracePeriod()
+	}
+	p.logger.WithField("grace termination period", gracePeriod).Warnf("The grace termination period is %v", gracePeriod)
```

Comment:
What does `.WithField().Warnf()` do for us? Is `logrus` or `p.logger` the correct logger we use?

Reply:
It add logrus entries to the warning message. Yes, we use pod.logger as a thin wrapper for logrus to provide extendable logs for p2.